src/main/java/com/iemr/tm/utils/JwtAuthenticationUtil.java

```diff
@@ -0,0 +1,91 @@
+package com.iemr.tm.utils;
+
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import com.iemr.tm.data.login.Users;
+import com.iemr.tm.repo.user.UserLoginRepo;
+import com.iemr.tm.utils.exception.IEMRException;
+
+import io.jsonwebtoken.Claims;
+import jakarta.servlet.http.HttpServletRequest;
+
+@Component
+public class JwtAuthenticationUtil {
+
+	@Autowired
+	private CookieUtil cookieUtil;
+	@Autowired
+	private JwtUtil jwtUtil;
+	@Autowired
+	private UserLoginRepo userLoginRepo;
+	private final Logger logger = LoggerFactory.getLogger(this.getClass().getName());
+
+	public JwtAuthenticationUtil(CookieUtil cookieUtil, JwtUtil jwtUtil) {
+		this.cookieUtil = cookieUtil;
+		this.jwtUtil = jwtUtil;
+	}
+
+	public ResponseEntity<String> validateJwtToken(HttpServletRequest request) {
+		Optional<String> jwtTokenOpt = cookieUtil.getCookieValue(request, "Jwttoken");
+
+		if (jwtTokenOpt.isEmpty()) {
+			return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+					.body("Error 401: Unauthorized - JWT Token is not set!");
+		}
+
+		String jwtToken = jwtTokenOpt.get();
+
+		// Validate the token
+		Claims claims = jwtUtil.validateToken(jwtToken);
+		if (claims == null) {
+			return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("Error 401: Unauthorized - Invalid JWT Token!");
+		}
+
+		// Extract username from token
+		String usernameFromToken = claims.getSubject();
+		if (usernameFromToken == null || usernameFromToken.isEmpty()) {
+			return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+					.body("Error 401: Unauthorized - Username is missing!");
+		}
+
+		// Return the username if valid
+		return ResponseEntity.ok(usernameFromToken);
+	}
+
+	public boolean validateUserIdAndJwtToken(String jwtToken) throws IEMRException {
+		try {
+			// Validate JWT token and extract claims
+			Claims claims = jwtUtil.validateToken(jwtToken);
+
+			if (claims == null) {
+				throw new IEMRException("Invalid JWT token.");
+			}
+
+			String userId = claims.get("userId", String.class);
+			String tokenUsername = jwtUtil.extractUsername(jwtToken);
+
+			// Fetch user based on userId from the database or cache
+			Users user = userLoginRepo.getUserByUserID(Long.parseLong(userId));
```

Comment:
It looks like only id and userName are compared.
In that case, other attributes are not required in the model.

Reply:
Fixed this, other attributes are removed